### PR TITLE
Feat/ab#75754 reduce white space backoffice when editing applications

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Reduced white space in back-office when editing applications, for this some changes were made:

- Instead of "back", put title of application
- put settings & permissions button next to application name ( when moved in blue header )
- put the publish button in the blue header as well
- removed the useless "close" button

## Useful links

[ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/75754)

## Type of change

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

It has been tested accessing some application and verifying if it's working correctly.

## Screenshots

Peek 25-09-2023 17-42

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MwanPygmay — APPROVED** [review]
read code, watched changes, looks good to me.

**Comment by AntoineRelief** [comment]
@RenzoPrats 
@MwanPygmay 

For future PRs,
also take into consideration how the UI is affected by the ticket
there, for example, the purpose of the ticket is to reduce the space used by the header, but as it was coded, the header is bigger than before..

Capture d’écran 2023-09-26 à 16 24 29

File: libs/shared/src/lib/components/layout/layout.component.ts [hunk 1]
```diff
@@ -43,14 +43,18 @@ export class LayoutComponent
   // === HEADER TITLE ===
   @Input() title = '';
 
+  @Input() settings: any[] = [];
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/layout/layout.component.ts]: @RenzoPrats 
cc @MwanPygmay 
comments !
whenever possible, add comments, even for class properties

**Comment by AntoineRelief** [comment]
@RenzoPrats 
@MwanPygmay 

unwanted issue created by the code:
Capture d’écran 2023-09-26 à 16 29 36

as you can see above, the permissions icon of the page is now white, and so, not visible!

**Review by AntoineRelief — CHANGES_REQUESTED** [review]
@RenzoPrats 
@MwanPygmay 
Will put this one back into progress, it's not working fine

**Comment by RenzoPrats** [comment]
@AntoineRelief @MwanPygmay I added coments in the new inputs, corrected the size of the header, and set back the grey color to the settings and permission buttons (but I guess white was more visible), the only persistent issue is about responsive, I'm not sure how it should work but for now the settings, permissions and publish button is being hidden as the page decreases size, if you @MwanPygmay can check please.

**Review by MwanPygmay — COMMENTED** [review]
- Comments were added
- Header size was corrected
- Permissions icon is not white anymore, although I am not sure about what you mean by 'as you can see above, the permissions icon of the page is now white, and so, not visible!' @AntoineRelief, to me grey is less visible but it presents the advantage of contrasting with the title

File: libs/shared/src/lib/components/layout/layout.component.ts (hunk 1 above)
Comment (AntoineRelief) [reply libs/shared/src/lib/components/layout/layout.component.ts]: @RenzoPrats 
jsdoc comments ;)
so starting with /**
and ending with */

**Comment by AntoineRelief** [comment]
@RenzoPrats 
cc @MwanPygmay 

You overcomplexified the ticket 
Indeed, the layout already had a property to inject a header template, so by putting the component inside the layout in order to display when no header was provided, you created a logic that could have been avoided

I changed that, by putting the application-toolbar component in the back-office, renaming it to application-header, changing the logic to be simplified, and then, injecting it as the header in the layout, so the layout does not need to import the component

Please keep that in mind for future tickets
Template injection is quite a powerful feature of Angular, that can simplify the code

**Pull request merged**